library/ic-ledger-types/src/lib.rs

```diff
@@ -123,6 +123,16 @@ impl fmt::Display for Tokens {
 )]
 pub struct Subaccount(pub [u8; 32]);
 
+impl From<Principal> for Subaccount {
+    fn from(principal: Principal) -> Self {
+        let mut subaccount = [0; 32];
+        let principal = principal.as_slice();
+        subaccount[0] = principal.len().try_into().unwrap();
+        subaccount[1..1 + principal.len()].copy_from_slice(principal);
+        Subaccount(subaccount)
```

Comment:
How about adding a test?  There are a few at the bottom of the file.

Reply:
Yeah I considered that but I didn't think of a suitable way to test this that isn't simply duplicating the logic in the test.
There are no existing functions exposed by `Subaccount` which could be used to verify the outcome of the `Principal` to `Subaccount` conversion.

Reply:
I could convert a `Principal` into a `Subaccount` and then check that the bytes match what is expected.
But the only way I'd be able to get the expected bytes is by running the function itself.

Reply:
Ok I've found a way that works and have added a unit test.